packages/cmf/__tests__/bootstrap.test.js

```diff
@@ -0,0 +1,180 @@
+import { render } from 'react-dom';
+import { routerMiddleware, syncHistoryWithStore } from 'react-router-redux';
+import createSagaMiddleware from 'redux-saga';
+
+import bootstrap, * as internals from '../src/bootstrap';
+import { registerInternals } from '../src/register';
+import actionCreator from '../src/actionCreator';
+import actions from '../src/actions';
+import { assertTypeOf } from '../src/assert';
+import component from '../src/component';
+import expression from '../src/expression';
+import storeAPI from '../src/store';
+import sagaRouter from '../src/sagaRouter';
+import sagas from '../src/sagas';
+
+jest.mock('react-dom', () => ({
+	render: jest.fn(),
+}));
+jest.mock('react-router-redux', () => ({
+	routerMiddleware: jest.fn(() => ({ routerMiddlewareMocked: true })),
+	syncHistoryWithStore: jest.fn((history, store) => ({ mockedSync: true, history, store })),
+}));
+jest.mock('redux-saga', () => {
+	const run = jest.fn();
+	const middleware = jest.fn(() => ({ reduxSagaMocked: true, run }));
+	middleware.run = run;
+	middleware.clearRun = () => run.mockClear();
+	return middleware;
+});
+
+// we mock all internal dependencies
+jest.mock('../src/actionCreator', () => ({
+	registerMany: jest.fn(),
+}));
+jest.mock('../src/component', () => ({
+	registerMany: jest.fn(),
+}));
+jest.mock('../src/expression', () => ({
+	registerMany: jest.fn(),
+}));
+jest.mock('../src/sagas', () => ({
+	registerMany: jest.fn(),
+}));
+jest.mock('../src/register', () => ({
+	registerInternals: jest.fn(),
+}));
+
+jest.mock('../src/store', () => ({
+	addPreReducer: jest.fn(),
+	setHttpMiddleware: jest.fn(),
+	setRouterMiddleware: jest.fn(),
+	initialize: jest.fn(() => ({ dispatch: jest.fn(), applyMiddleware: jest.fn(), })),
+}));
+
+describe('bootstrap', () => {
+	describe('bootstrap registry', () => {
+		it('should check options', () => {
+			const toThrow = () => bootstrap({ appId: {} });
+			expect(toThrow).toThrow('appId must be a string but got object');
+		});
+		it('should call registerInternals', () => {
+			registerInternals.mockClear();
+			bootstrap({});
+			expect(registerInternals).toHaveBeenCalled();
+		});
+		it('should registery options.components using component.registerMany', () => {
+			component.registerMany.mockClear();
+			const options = {
+				components: {
+					foo: jest.fn(),
+				},
+			};
+			bootstrap(options);
+			expect(component.registerMany).toHaveBeenCalledWith(options.components);
+		});
+		it('should registery options.expressions using expression.registerMany', () => {
+			expression.registerMany.mockClear();
+			const options = {
+				expressions: {
+					foo: jest.fn(),
+				},
+			};
+			bootstrap(options);
+			expect(expression.registerMany).toHaveBeenCalledWith(options.expressions);
+		});
+		it('should support options.actionCreators', () => {
```

Comment:
Can you align the description with the other tests please ? `should register options.actionCreators using actionCreator.registerMany`